internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -0,0 +1,397 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kubernetessecret
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	secretexportmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster/secretexport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kubernetessecret/scope"
+)
+
+func ResourceSecret() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceSecretRead,
+		CreateContext: resourceSecretCreate,
+		DeleteContext: resourceSecretDelete,
+		Schema:        secretSchema,
+	}
+}
+
+var secretSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	NamespaceNameKey: {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	OrgIdKey: {
+		Type: schema.TypeString,
+	},
+	scope.ScopeKey: scope.ScopeSchema,
+	specKey:        secretSpec,
+	ExportKey: {
+		Type:     schema.TypeBool,
+		Required: true,
+	},
+	statusKey: {
+		Type:     schema.TypeMap,
+		Computed: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	},
+}
+
+type dockerConfigJSON struct {
+	Auths map[string]map[string]interface{} `json:"auths,omitempty"`
+}
+
+var secretSpec = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the kubernetes secret",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			DockerConfigjsonKey: {
+				Type:        schema.TypeList,
+				Description: "SecretType definition - SECRET_TYPE_DOCKERCONFIGJSON, Kubernetes secrets type.",
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						UsernameKey: {
```

Comment:
Similarly, define optional/required/computed etc and other schema components for each key.

Reply:
added